pkg/modules/mockmodules/internal/impl.go

```diff
@@ -0,0 +1,10 @@
+package internal
+
+import (
+	"github.com/filecoin-project/mir/pkg/events"
+	"github.com/filecoin-project/mir/pkg/pb/eventpb"
+)
+
+type ModuleImpl interface {
+	Event(ev *eventpb.Event) (*events.EventList, error)
```

Comment:
I wonder how well `gomock` matches `events.EventList`. Would you mind to try matching something other than `events.EmptyList()`?

Reply:
Afaik, it doesn't need to match the return value, only the parameters (in this case, an event).

But in case you are still wondering, it does seem to work as expected with some non-empty event lists (see [node_test.go](https://github.com/filecoin-project/mir/blob/main/node_test.go), for example). Internally, it simply uses `reflect.DeepEqual`.

However, it doesn't work well with protobufs in certain cases since they may have some nasty unexported fields.
I'll patch it up a bit in another PR.